client/js/editmode.js

```diff
@@ -1,16 +1,23 @@
 let edit = false;
 
-function generateUniqueWidgetID() {
+function generateUniqueWidgetID(widget) {
   let id;
+  let i = 1;
   do {
-    id = Math.random().toString(36).substring(3, 7);
+    if(widget && widget.type == 'card')
+      id = widget.deck + ' - ' + widget.cardType + ' - ' + i;
+    else if(widget)
+      id = (widget.type || 'basic') + ' - ' + i;
```

Comment:
Including 'basic' is going to be more annoying then useful. The base widget for building stuff will need to be constantly renames.
I recommend adding more specific names for game pieces. This can be done directly into each game pice code.